python-attr-lookup/src/plang/PythonObject.java

```diff
@@ -51,7 +51,10 @@ public List<PythonObject> getMRO() {
      * result (i.e. it remembers the list buildMRO() returned and keeps returning it).
      */
     protected List<PythonObject> buildMRO() {
-        throw new UnsupportedOperationException("not implemented yet");
+        mro = new ArrayList<PythonObject>();
```

Comment:
The assumption with this is that buildMRO() is only ever called by getMRO(), which I did not realize. As a result, I set `mro` in here instead of using a separate variable. It won't cause any problems, but it's a bit weird.

Reply:
The side effect here does make the code a bit brittle. You’re creating a brief window where the `mro` var is a mutable collection, then immediately gets replaced with an immutable one — and making it possible to leave the mutable one there permanently by calling `buildMRO` directly from somewhere else. I agree that won’t cause any problems as the code exists _now_, but it does open up a window for future bugs.